### PR TITLE
Fix Issue #684

### PR DESCRIPTION
**Pull request by jday1**
See https://github.com/airr-community/airr-standards/issues/684 for context.

Introduced GenotypeSets to allow testing of mhc_genotype_set and receptor_genotype_set.

Other minor bug fixes.

### CODE REVIEW

File: lang/R/inst/extdata/airr-schema.yaml
Comment (jday1) [inline lang/R/inst/extdata/airr-schema.yaml:41]: There are several cases where there is an indentation of 2 is used instead of 4. This lack of consistency can cause some validation to fail. For example, in the case of the MHC enum on line 1273. A long term solution is to enforce yamls are written consistently in the github action checks to prevent this.

File: lang/R/inst/extdata/airr-schema.yaml
Comment (jday1) [inline lang/R/inst/extdata/airr-schema.yaml:825]: Ignore: My IDE yaml formatter did this.

File: lang/R/inst/extdata/airr-schema.yaml
Comment (jday1) [inline lang/R/inst/extdata/airr-schema.yaml:2367]: Ignore: My IDE yaml formatter did this.

File: lang/R/inst/extdata/airr-schema.yaml
Comment (jday1) [inline lang/R/inst/extdata/airr-schema.yaml:2383]: Ignore: My IDE yaml formatter did this.

File: lang/R/inst/extdata/airr-schema.yaml
Comment (jday1) [inline lang/R/inst/extdata/airr-schema.yaml:4123]: Ignore: My IDE yaml formatter did this.

File: lang/R/inst/extdata/airr-schema.yaml
Comment (jday1) [inline lang/R/inst/extdata/airr-schema.yaml:4136]: Ignore: My IDE yaml formatter did this.

File: lang/R/inst/extdata/airr-schema.yaml
Comment (jday1) [inline lang/R/inst/extdata/airr-schema.yaml:4369]: Ignore: My IDE yaml formatter did this.

File: lang/python/airr/specs/airr-schema.yaml
Comment (jday1) [inline lang/python/airr/specs/airr-schema.yaml:825]: Ignore: My IDE yaml formatter did this.

File: lang/python/airr/specs/airr-schema.yaml
Comment (jday1) [inline lang/python/airr/specs/airr-schema.yaml:2367]: Ignore: My IDE yaml formatter did this.

File: lang/python/airr/specs/airr-schema.yaml
Comment (jday1) [inline lang/python/airr/specs/airr-schema.yaml:2383]: Ignore: My IDE yaml formatter did this.

File: lang/python/airr/specs/airr-schema.yaml
Comment (jday1) [inline lang/python/airr/specs/airr-schema.yaml:4123]: Ignore: My IDE yaml formatter did this.

File: lang/python/airr/specs/airr-schema.yaml
Comment (jday1) [inline lang/python/airr/specs/airr-schema.yaml:4136]: Ignore: My IDE yaml formatter did this.

File: lang/python/airr/specs/airr-schema.yaml
Comment (jday1) [inline lang/python/airr/specs/airr-schema.yaml:4369]: Ignore: My IDE yaml formatter did this.

File: lang/python/tests/data/bad_germline_set.json
Comment (jday1) [inline lang/python/tests/data/bad_germline_set.json:351]: Ignore: My IDE yaml formatter did this.

File: lang/python/tests/data/good_combined_airr.json
Comment (jday1) [inline lang/python/tests/data/good_combined_airr.json:62]: Adding a test example.

File: lang/python/tests/data/good_combined_airr.json
Comment (jday1) [inline lang/python/tests/data/good_combined_airr.json:894]: Ignore: My IDE json formatter did this.

**Comment by javh** [comment]
From the call:
* Rename `GenotypeSets` to `SubjectGenotype`.
* Fine to merge for v1.4.2 after testing on iReceptor, with the understanding that the long term solution will differ.

**Comment by bcorrie** [comment]
These queries will test the deepest level objects in the MHC Genotype so is a good test to make sure nesting changes of objects doesn't break the structure.

curl -d '{"filters":{"op":"=", "content": { "field":"subject.genotype.mhc_genotype_set.mhc_genotype_list.mhc_alleles.gene.id", "value":"MRO:0000051"}}}' https://t1d-1.ireceptor.org/airr/v1/repertoire

curl -d '{"filters":{"op":"=", "content": { "field":"subject.genotype.mhc_genotype_set.mhc_genotype_list.mhc_alleles.allele_designation", "value":"HLA-DPB1*15"}}}' https://t1d-1.ireceptor.org/airr/v1/repertoire

**Review by javh — APPROVED** [review]
Looks good to me, assuming it passes the iReceptor checks.

**Review by bcorrie — APPROVED** [review]
Nested objected definition looks fine.

**Comment by javh** [comment]
From the call:
* Good to merge after resolving merge conflicts.

**Comment by javh** [comment]
Moving this to #695, because it was less effort to fix the merge conflicts...

**Comment by javh** [comment]
Merged via #695.

**Pull request closed without merging**
